quickstart/101-batch-account-with-storage/variables.tf

```diff
@@ -0,0 +1,17 @@
+variable "resource_group_location" {
+  type        = string
+  default     = "eastus"
+  description = "Location for all resources."
+}
+
+variable "resource_group_name_prefix" {
+  type        = string
+  default     = "rg"
+  description = "Prefix of the resource group name that's combined with a random ID so name is unique in your Azure subscription."
+}
+
+variable "storage_account_type" {
```

Comment:
This needs to have a validation to ensure the string is one of the allowed values and is in the format that can be maniuplated by config. See custom_conditions for more info: https://developer.hashicorp.com/terraform/language/expressions/custom-conditions

Reply:
A `validation` block like this could finish the job:

```hcl
validation {
    condition = can(split(var.storage_account_type, "_")[1])
    error_message = "invalid storage account type, should be like `Standard_LRS`."
}
```

Reply:
This validation will prevent the string manipulation from failing but since the allowed values are known, I was thinking of something more like this:
```terraform
validation {
    condition = contains(["Premium_LRS","Premium_ZRS","Standard_GRS","Standard_GZRS","Standard_LRS","Standard_RAGRS","Standard_RAGZRS","Standard_ZRS"], var.storage_account_type)
    error_message = "Invalid storage account type. The value should be one of the following: 'Premium_LRS','Premium_ZRS','Standard_GRS','Standard_GZRS','Standard_LRS','Standard_RAGRS','Standard_RAGZRS','Standard_ZRS'."
}
```
Known values: https://learn.microsoft.com/en-us/azure/templates/microsoft.storage/storageaccounts?pivots=deployment-language-terraform#sku-2